### ISSUE TITLE
Rule Change: (fill in)

### ISSUE THREAD

**Issue by kevinbarabash** [eslint/eslint#15199]
### What rule do you want to change?

no-undef

### What change to do you want to make?

Generate fewer warnings

### How do you think the change should be implemented?

A new default behavior

### Example code

```js
globalThis.foo = "foo";
```

### What does the rule currently do for this code?

It complains about `globalThis` not being defined.

### What will the rule do after it's changed?

`globalThis` will be allowed by default.

### Participation

- [X] I am willing to submit a pull request to implement this change.

### Additional comments

The `globalThis` proposal is now at stage-4, see https://github.com/tc39/proposal-global.  This issue was previously reported in https://github.com/eslint/eslint/issues/14639, but at the time the proposal as at stage-3.
